lib/classes/Logger.js

```diff
@@ -0,0 +1,211 @@
+ * @param level The log level
+ * @returns {Function} the logging function
+ */
+function buildLogMethod(level) {
+  // the log method takes a message, and any extra styles to apply (e.g., underline, dim, etc.)
+  return function () {
+    const message = _.head(arguments);
+    const extraStyles = _.tail(arguments);
+    const styleArr = [this.styles[level]];
+
+    // if extra styles were passed in, we need to map them appropriately and add to the styleArr
+    if (Array.isArray(extraStyles) && extraStyles.length > 0) {
+      extraStyles.forEach((s) => {
+        if (this.styles[s]) {
+          styleArr.push(this.styles[s]);
+        } else {
+          styleArr.push(s);
+        }
+      });
+    }
+
+    // call internal logger fn
+    logInternal(message, styleArr.join('.'));
+
+    // return `this` for method chaining
+    return this;
+  };
+}
+
+/**
+ * Provides console logging functionality with common styling
+ */
+class Logger {
+
+  constructor(mappingOverride) {
+    this.styles = {};
+
+    // apply default style mapping, then override with passed in config (if present)
+    _.defaults(this.styles, mappingOverride || {}, styleMapping);
+  }
+
+  /**
+   * Get the style mapping currently in use
+   * @return {{}|*}
+   */
+  getStyles() {
+    return _.clone(this.styles);
+  }
+
+  /**
+   * Convenience method to get a single style mapping
+   * @param key the key
+   * @return {*} the style value
+   */
+  getStyle(key) {
+    return this.styles[key];
+  }
+
+  /**
+   * Apply styles to the given message
+   * @param message the message to style
+   * @param styles... the styles to use (e.g., BOLD, UNDERLINE, etc.)
+   * @return {String} the styled message
+   */
+  style() {
+    const message = _.head(arguments);
+    const extraStyles = _.tail(arguments);
+
+    if (Array.isArray(extraStyles) && extraStyles.length > 0) {
+      return applyStyle(message, extraStyles.join('.'));
+    }
+
+    return applyStyle(message);
+  }
+
+  /**
+   * Convenience method to log a newline on the console
+   */
+  newline() {
+    return this.log('');
+  }
+}
+
+/**
+ * Log message using default style
+ * @type {Function}
+ */
+Logger.prototype.log = buildLogMethod('log');
+
+/**
+ * Log message using 'info' style
+ * @type {Function}
+ */
+Logger.prototype.info = buildLogMethod('info');
+
+/**
+ * Log message using 'debug' style
+ * @type {Function}
+ */
+Logger.prototype.debug = buildLogMethod('debug');
+
+/**
+ * Log message using 'warn' style
+ * @type {Function}
+ */
+Logger.prototype.warn = buildLogMethod('warn');
+
+/**
+ * Log message using 'error' style
+ * @type {Function}
+ * @param {String} message The message to log
+ * @param {String...} styles The extra styles to apply
+ */
+Logger.prototype.error = buildLogMethod('error');
+
+
+// apply all style properties to the logger prototype as uppercase property names
+Object.defineProperties(Logger.prototype, (() => {
```

Comment:
You can use `.reduce` instead of `.forEach` here, save a few lines and avoid the IIFE.